monai/apps/reconstruction/mri_utils.py

```diff
@@ -0,0 +1,246 @@
+    are stacked along the last dimension.
+    inputs:
+        data (np.array): Input numpy array
+    outputs:
+        torch.Tensor: PyTorch version of data
+    """
+    if np.iscomplexobj(data):
+        data = np.stack((data.real, data.imag), axis=-1)
+    return monai.utils.type_conversion.convert_to_tensor(data)
+
+
+def complex_abs(x: ndarray) -> ndarray:
+    """
+    Compute the absolute value of a complex array.
+    inputs:
+        x (np.array): Input numpy array with 2 channels in the last
+        dimension representing real and imaginary parts.
+    outputs:
+        np.array: Absolute value along the last dimention
+    """
+    assert x.shape[-1] == 2
+    return np.sqrt(x[..., 0] ** 2 + x[..., 1] ** 2)
+
+
+# mask functions
+class MaskFunc:
+    def __init__(self, center_fractions: list, accelerations: list) -> None:
+        """
+        inputs:
+            center_fractions (List[float]): Fraction of low-frequency columns to be retained.
+                If multiple values are provided, then one of these numbers is chosen uniformly
+                each time.
+
+            accelerations (List[int]): Amount of under-sampling. This should have the same length
+                as center_fractions. If multiple values are provided, then one of these is chosen
+                uniformly each time.
+        """
+        if len(center_fractions) != len(accelerations):
+            raise ValueError("Number of center fractions should match number of accelerations")
+
+        self.center_fractions = center_fractions
+        self.accelerations = accelerations
+        self.rng = np.random.RandomState()
+
+    def choose_acceleration(self):
+        choice = self.rng.randint(0, len(self.accelerations))
+        center_fraction = self.center_fractions[choice]
+        acceleration = self.accelerations[choice]
+        return center_fraction, acceleration
+
+
+def create_mask_for_mask_type(mask_type_str: str, center_fractions: list, accelerations: list) -> MaskFunc:
+    """
+    Create an under-sampling mask generator
+    inputs:
+        mask_type_str (string): denotes the mask type ('random','equispaced')
+        center_fractions (List[float]): Fraction of low-frequency columns to be retained.
+        If multiple values are provided, then one of these numbers is chosen uniformly each time.
+        accelerations (List[int]): Amount of under-sampling. This should have the same length as center_fractions.
+        If multiple values are provided, then one of these is chosen uniformly each time.
+    outputs:
+        callable mask function
+    """
+    if mask_type_str == "random":
+        return RandomMaskFunc(center_fractions, accelerations)
+    elif mask_type_str == "equispaced":
+        return EquispacedMaskFunc(center_fractions, accelerations)
+    else:
+        raise Exception(f"{mask_type_str} not supported")
+
+
+class RandomMaskFunc(MaskFunc):
```

Comment:
Is this used as a preprocessing transform? If so, may be better to put it in transform rather than utils. We hope utils to be funcs, not class.
Thanks!

Reply:
Got it.